plugins/GSdx/GSUtil.cpp

```diff
@@ -396,6 +396,41 @@ D3D_FEATURE_LEVEL GSUtil::CheckDirect3D11Level(IDXGIAdapter *adapter, D3D_DRIVER
 	return SUCCEEDED(hr) ? level : (D3D_FEATURE_LEVEL)0;
 }
 
+GSRendererType GSUtil::GetBestRenderer()
+{
+	CComPtr<IDXGIAdapter1> adapter;
+	DXGI_ADAPTER_DESC1 desc;
+	CComPtr<IDXGIFactory1> dxgi_factory;
+	GSRendererType BestRenderer = GSRendererType::Default;
+	bool Result = false;
+
+	if (GSUtil::CheckDXGI())
+	{
+		Result = SUCCEEDED(CreateDXGIFactory1(IID_PPV_ARGS(&dxgi_factory)));
+	}
+
+	if(GSUtil::CheckD3D11() && Result)
```

Comment:
> A suggestion: maybe use early return instead? Something like:

I like the first one. After that,  I used consecutive ``SUCCEEDED()`` conditional checks instead. does it look fine to you ?